olc_art.cpp

```diff
@@ -869,41 +867,28 @@ olc_db::try_get_result_type olc_db::try_get(detail::art_key k) const noexcept {
     return {};
     // LCOV_EXCL_STOP
   }
-
-  auto node{root.load()};
-
-  if (UNODB_DETAIL_UNLIKELY(node == nullptr)) {
-    if (UNODB_DETAIL_UNLIKELY(!parent_critical_section.try_read_unlock())) {
-      // LCOV_EXCL_START
-      spin_wait_loop_body();
-      return {};
-      // LCOV_EXCL_STOP
-    }
+  detail::olc_node_ptr node{root.load()};
+  if (UNODB_DETAIL_UNLIKELY(node == nullptr)) {  // special path if empty tree.
+    if (UNODB_DETAIL_UNLIKELY(!parent_critical_section.try_read_unlock()))
+      return {};  // LCOV_EXCL_LINE
+    // return an empty result (breaks out of caller's while(true) loop)
     return std::make_optional<get_result>(std::nullopt);
   }
 
   auto remaining_key{k};
-
-  if (UNODB_DETAIL_UNLIKELY(!parent_critical_section.check())) {
-    // LCOV_EXCL_START
```

Comment:
This is another substantive change @laurynas-biveinis.

Reply:
This is the issue. This check protects root changes to and from `nullptr`, and removing this check exposes us to the race between creating a new root and reading from it - exactly what we saw